internal/services/iothub/iothub_resource.go

```diff
@@ -611,20 +610,15 @@ func resourceIotHubCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) err
 	stateConf := &pluginsdk.StateChangeConf{
 		Pending: []string{"Activating", "Transitioning"},
 		Target:  []string{"Succeeded"},
-		Refresh: iothubStateRefreshFunc(ctx, client, resourceGroup, name),
+		Refresh: iothubStateRefreshFunc(ctx, client, id.ResourceGroup, id.Name),
 		Timeout: d.Timeout(timeout),
 	}
 
 	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
-		return fmt.Errorf("waiting for the completion of the creating/updating of IotHub %q (Resource Group %q): %+v", name, resourceGroup, err)
-	}
-
-	resp, err := client.Get(ctx, resourceGroup, name)
-	if err != nil {
-		return err
```

Comment:
is there a reason we are removing the 2nd get here?

Reply:
It looked like the reason we were performing a get here was to set the id attribute, but this is now set with the ID generated by the parser. If this get also serves as a check that the resource was created properly I can put it back in and throw away the response with _ (otherwise it makes the linter cry)